static/js/controllers/reports.js

```diff
@@ -369,13 +461,74 @@ var _ = require('underscore'),
         });
       };
       $scope.resetFilterModel = function() {
+        if ($scope.selectMode && $scope.selected && $scope.selected.length) {
+          // can't filter when in select mode
+          return;
+        }
         $scope.filters = {};
         SearchFilters.reset();
         $scope.search();
       };
 
       $scope.search();
 
+      $('.inbox').on('click', '#reports-list .message-wrapper', function(e) {
+        if ($scope.selectMode) {
+          e.preventDefault();
+          e.stopPropagation();
+          var target = $(e.target).closest('li');
+          var reportId = target.attr('data-record-id');
+          var checkbox = target.find('input[type="checkbox"]');
+          var alreadySelected = _.findWhere($scope.selected, { _id: reportId });
+          // timeout so if the user clicked the checkbox it has time to
+          // register before we set it to the correct value.
```

Comment:
When a user clicks on a checkbox the browser toggles it between checked and unchecked. This is bad in this case because we want to control the checking and unchecking. One solution is to let the browser do its thing and then immediately set it the way we want it set. This is achieved by letting the event complete and then running our checked state setting code. The "timeout" is putting our code at the end of the execution queue so it'll run immediately after the browser has finished doing the wrong thing.